platform/wayland/cog-platform-wl.c

```diff
@@ -2125,108 +2339,44 @@ clear_egl (void)
 }
 
 static gboolean
-create_window (GError **error)
+create_window(GError **error)
 {
-    g_debug ("Creating Wayland surface...");
+    g_debug("Creating Wayland surface...");
 
-    win_data.wl_surface = wl_compositor_create_surface (wl_data.compositor);
-    g_assert (win_data.wl_surface);
+    win_data.shell_context->wl_shell_data.wl_surface = wl_compositor_create_surface(wl_data.compositor);
```

Comment:
Why has wl_surface been moved to wl_shell_data? It doesn't seem to be specific to wl_shell and putting it here means that you're going to corrupt data in some way or another when using any shell that isn't wl_shell.